pom.xml

```diff
@@ -36,35 +36,55 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
     <main.basedir>${project.basedir}</main.basedir>
-    <main.java.version>1.8</main.java.version>
-    <main.signature.artifact>java18</main.signature.artifact>
+
+    <!-- All source is Java 11+ -->
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <!-- We don't use animal-sniffer anymore as release obviates it.
+         See https://github.com/mojohaus/animal-sniffer/issues/62 -->
+    <maven.compiler.release>11</maven.compiler.release>
 
     <!-- override to set exclusions per-project -->
     <errorprone.args />
-    <errorprone.version>2.23.0</errorprone.version>
+    <errorprone.version>2.24.0</errorprone.version>
 
-    <!-- TODO: can't update to 2.13 without updating ES driver -->
+    <!-- Use a common scala binary version, and the highest spark patch version.
+         Scala 2.13 is held back due to spark version conflicts until Elastic
+         updates their version. Once 2.13 works, we can also use JDK 17 per
```

Comment:
Theoretically, you can get scala 2.12 to work on later JREs, but SPARK-33772 says it is not advised and to update to 2.13 first